src/tools/perf/lib/ucp_tests.cc

```diff
@@ -725,6 +725,39 @@ class ucp_perf_test_runner {
         wait_send_window(m_max_outstanding);
     }
 
+    bool use_ack() const
+    {
+        /* TODO: daemon does not support bi-directional flow */
+        if (m_perf.params.ucp.is_daemon_mode) {
+            return false;
+        }
+
+        return (CMD == UCX_PERF_CMD_TAG) || (CMD == UCX_PERF_CMD_TAG_SYNC) ||
+               (CMD == UCX_PERF_CMD_AM);
+    }
+
+    void send_ack(ucp_ep_h ep, void *buffer, ucp_datatype_t datatype,
+                  uint64_t remote_addr, ucp_rkey_h rkey)
+    {
+        if (!use_ack()) {
+            return;
+        }
+
+        send(ep, buffer, 1, datatype, 0, remote_addr, rkey, false);
```

Comment:
shall we also progress this side with maybe `wait_send_window()`, in case progress would at some point happen to be needed, or the flush line 929 would be covering it?

Reply:
I had `wait` initially but then removed it, because flush must be doing the job
But I can add it back, it does not harm anyway

Reply:
added.
It was actually a reason of CI test failure